flax/linen/recurrent.py

```diff
@@ -726,12 +759,143 @@ def _select_last(sequence: A, segmentation_mask: jnp.ndarray, axis: int) -> A:
   last_idx = segmentation_mask.sum(axis=-1) - 1
 
   def _slice_array(x: jnp.ndarray):
-    _last_idx = _expand_dims_for(last_idx, to_target=x)
+    _last_idx = _expand_dims_like(last_idx, target=x)
     x = jnp.take_along_axis(x, _last_idx, axis=axis)
     return x.squeeze(axis=axis)
 
   return jax.tree_map(_slice_array, sequence)
 
-def _expand_dims_for(x, *, to_target):
+def _expand_dims_like(x, *, target):
```

Comment:
Nit: I am not sure why `target` is required to be a kwarg. I think it is also fine as a positional argument.

Reply:
changed to regular arg